docs/references.md

```diff
@@ -0,0 +1,326 @@
+# Working with References
```

Comment:
Make sure this document is update to date and true for the features we've implemented thus far.